feat(blog): add 1.1.0 announcement post (#899)
tauri-apps/tauri-docs: blog/2022-09-15-tauri-1.1.mdx

---
authors: [lucasfernog]
image: ./tauri_1_1_images/header.png
description: After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release.
---

# Announcing Tauri 1.1.0

![Tauri 1.0 Launch Hero Image](./tauri_1_1_images/header.png)

After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release. The changes were internally audited and no security issues were found.

You can update the NPM dependencies with:

```
$ npm update
```

To update the Cargo dependencies, run the following in the `src-tauri` folder:

```
$ cargo update
```

## What's in 1.1.0

### Security patch

> This release includes a patch for a security vulnerability reported by martin-ocasek. The `readDir` function was able to return entries outside the configured scope when a symlink is found. The patch is also available in 1.0.6. See [the issue](https://github.com/tauri-apps/tauri/issues/4882) for more details.

Reviewer: ```suggestion
This release includes a patch for a security vulnerability reported by [@martin-ocasek](https://github.com/martin-ocasek). The `readDir` function was able to return entries outside the configured scope when a symlink is found. The patch is also available in 1.0.6. See [the issue](https://github.com/tauri-apps/tauri/issues/4882) for more details.
```

### Icon generation

We have been recommending using the [tauricon](https://github.com/tauri-apps/tauricon) project to generate icons for your Tauri application using a single source PNG. Several issues have been reported, and we decided to `Rewrite It In Rust` to enhance its stability. This allowed us to move this functionality to the main Tauri CLI, so now you can use the `tauri icon` command.

### cargo-binstall support for the Tauri CLI

The Tauri CLI can now be installed using [cargo-binstall](https://github.com/cargo-bins/cargo-binstall), a mechanism to download and install prebuilt Rust binaries. The binaries are available for the main targets and can be installed with:

```

> $ cargo install cargo-binstall

Reviewer: maybe remove this and let them follow `cargo-binstall` readme for installation?

Author: this is my bad LOL should be just `cargo binstall tauri-cli` but i made a mistake in the CLI cargo.toml metadata for binstall :(

Reviewer: then maybe add a note why it should be installed like this until the next release

```json
tauri.conf.json:
{
  "build": {
      "devPath": "http://localhost:8000",
      "distDir": "../dist"
  }
}
```

```json5
tauri.conf.json5:
{
  build: {
      // devServer URL (comments are allowed!)
      devPath": http://localhost:8000,
      distDir": ../dist
  }
}
```

The 1.1.0 release includes TOML support behind the `config-toml` Cargo feature. Now you can define your Tauri configuration in a `Tauri.toml` file:

```toml
[build]
dev-path = "http://localhost:8000"
dist-dir = "../dist"
```

### Dependency updates

This release includes some dependency updates that must be handled in your app if you implement platform-specific functionalities using these crates. The most important updates are:

- `windows` updated to 0.39.0
- `webview2-com` updated to 0.19.1
- `raw-window-handle` updated to 0.5.0

Make sure you also update plugins such as `window-vibrancy` and `window-shadows` to latest.

## Contributors to 1.1.0

The Tauri team thanks the following contributors for the 1.1.0 release:

> - Berrysoft
> - keraf
> - jsoref
> - A-kirami
> - olivierlemasle
> - mateo-gallardo
> - rockerBOO
> - Flecart
> - brian14708
> - koriwi
> - paul-soporan
> - shniubobo
> - horochx
> - RubenKelevra
> - LIMPIX31
> - AxlLind

Reviewer: ```suggestion
- [Berrysoft](https://github.com/Berrysoft)
- [keraf](https://github.com/keraf)
- [jsoref](https://github.com/jsoref)
- [A-kirami](https://github.com/A-kirami)
- [olivierlemasle](https://github.com/olivierlemasle)
- [mateo-gallardo](https://github.com/mateo-gallardo)
- [rockerBOO](https://github.com/rockerBOO)
- [Flecart](https://github.com/Flecart)
- [brian14708](https://github.com/brian14708)
- [koriwi](https://github.com/koriwi)
- [paul-soporan](https://github.com/paul-soporan)
- [shniubobo](https://github.com/shniubobo)
- [horochx](https://github.com/horochx)
- [RubenKelevra](https://github.com/RubenKelevra)
- [LIMPIX31](https://github.com/LIMPIX31)
- [AxlLind](https://github.com/AxlLind)
```

> ---

Reviewer 2: I'd rename the file to be 1-1 instead of 1.1 (similar to the 1-0 filename)

authors: [lucasfernog]
image: ./tauri_1_1_images/header.png
description: After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release.
---

# Announcing Tauri 1.1.0

![Tauri 1.0 Launch Hero Image](./tauri_1_1_images/header.png)

> After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release. The changes were internally audited and no security issues were found.

Reviewer 2: ```suggestion
After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the [1.1.0 release](https://github.com/tauri-apps/tauri/releases/tag/tauri-v1.1.0). The changes were internally audited and no security issues were found.
```

> After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release. The changes were internally audited and no security issues were found.
> 
> You can update the NPM dependencies with:

Reviewer 2: Need to apply this to other posts as well, but makes it better on the list page

```suggestion
After 113 pull requests and nearly two months of work, the Tauri team is pleased to announce the 1.1.0 release. The changes were internally audited and no security issues were found.

<!--truncate-->

You can update the NPM dependencies with:
```

```
$ npm update

> ```

Reviewer 2: Do you want me to add a command tab switcher thing to this to include Yarn and pnpm? Is it `yarn upgrade` and `pnpm update`?

Reviewer 3: we could also link to the "update dependencies" guide tho that one needs changes too :joy:

To update the Cargo dependencies, run the following in the `src-tauri` folder:

```
$ cargo update
```

> ## What's in 1.1.0

Reviewer 2: ```suggestion
## What's New in 1.1.0
```

### Security patch

> This release includes a patch for a security vulnerability reported by [@martin-ocasek](https://github.com/martin-ocasek). The `readDir` function was able to return entries outside the configured scope when a symlink is found. The patch is also available in 1.0.6. See [the issue](https://github.com/tauri-apps/tauri/issues/4882) for more details.

Reviewer 2: ```suggestion
This release includes a patch for a security vulnerability reported by [@martin-ocasek](https://github.com/martin-ocasek). The `readDir` function was able to return entries outside the configured scope when a symlink is found. The patch is also available in Tauri [1.0.6](https://github.com/tauri-apps/tauri/releases/tag/tauri-v1.0.6). See [the issue on GitHub](https://github.com/tauri-apps/tauri/issues/4882) for more details.
```

> ### Icon generation

Reviewer 2: ```suggestion
### Icon Generation
```

> We have been recommending using the [tauricon](https://github.com/tauri-apps/tauricon) project to generate icons for your Tauri application using a single source PNG. Several issues have been reported, and we decided to `Rewrite It In Rust` to enhance its stability. This allowed us to move this functionality to the main Tauri CLI, so now you can use the `tauri icon` command.

Reviewer 2: ```suggestion
We have been recommending to use the [tauricon](https://github.com/tauri-apps/tauricon) project to generate icons for your Tauri application using a single source PNG. Several issues have been reported, and we decided to "Rewrite It In Rust" to enhance its stability. This allowed us to move this functionality to the main Tauri CLI, so now you can use the [`tauri icon`](https://tauri.app/v1/api/cli#icon) command.
```

> ### cargo-binstall support for the Tauri CLI

Reviewer 2: ```suggestion
### `cargo-binstall` Support for Tauri CLI
```

> The Tauri CLI can now be installed using [cargo-binstall](https://github.com/cargo-bins/cargo-binstall), a mechanism to download and install prebuilt Rust binaries. The binaries are available for the main targets and can be installed with:

Reviewer 2: ```suggestion
The Tauri CLI can now be installed using [cargo-binstall](https://github.com/cargo-bins/cargo-binstall), a mechanism to download and install pre-built Rust binaries. The binaries are available for the main targets and can be installed with:
```

```
$ cargo install cargo-binstall
# in the next release, `$ cargo binstall tauri-cli` will be enough
$ cargo binstall tauri-cli --pkg-url "{ repo }/releases/download/cli.rs-v{ version }/cargo-tauri-{ target }.{ archive-format }"

> $ cargo tauri dev # run any tauri command!

Reviewer 2: ```suggestion
$ cargo tauri dev # run any Tauri command!
```

```

> ### Create system trays at runtime

Reviewer 2: ```suggestion
### Create System Trays at Runtime
```

> The system tray APIs previously only available in `tauri::Builder::system_tray` function can now be used at runtime, giving you control over its lifetime and even create multiple trays.

Reviewer 2: ```suggestion
The system tray APIs (previously only available in `tauri::Builder::system_tray`) can now be used at runtime with [`tauri::SystemTray`](https://docs.rs/tauri/1.1.0/tauri/struct.SystemTray.html) giving you control over its lifetime and even create multiple trays.
```

Here's a quick example on how to use it:

```rust
use tauri::{Builder, CustomMenuItem, SystemTray, SystemTrayEvent, SystemTrayMenu};
Builder::default()
    .setup(|app| {
        let handle = app.handle();
        SystemTray::new()
            .with_id("main")
            .with_menu(
                SystemTrayMenu::new().add_item(CustomMenuItem::new("quit", "Quit"))
            )
            .on_event(move |event| {
                let tray_handle = handle.tray_handle_by_id("main").unwrap();
                if let SystemTrayEvent::MenuItemClick { id, .. } = event {
                    if id == "quit" {
                        tray_handle.destroy().unwrap();
                    }
                }

>             })

Reviewer 3: ```suggestion
            })
            .build(&handle)
            .expect("unable to create tray");
```

    });
```

> ### TOML configuration support

Reviewer 2: ```suggestion
### TOML Configuration Support
```

In the 1.0 releases Tauri supports the `JSON` configuration format by default, and `JSON5` when the `config-json5` Cargo feature is enabled, meaning the following configurations are valid:

> ```json
> tauri.conf.json:

Reviewer 3: ```suggestion
```json title=tauri.conf.json
```

Reviewer 3: see lorenzo's review

Reviewer 2: copycat (even though you clicked submit first)

{
  "build": {
      "devPath": "http://localhost:8000",
      "distDir": "../dist"
  }
}

> ```

Reviewer 2: ~~~suggestion
```json title=tauri.conf.json
{
  "build": {
      "devPath": "http://localhost:8000",
      "distDir": "../dist"
  }
}
```
~~~

> ```json5
> tauri.conf.json5:

Reviewer 3: ```suggestion
```json5 title=tauri.conf.json5
```

Reviewer 3: same

{
  build: {
      // devServer URL (comments are allowed!)
      devPath": http://localhost:8000,
      distDir": ../dist
  }
}

> ```

Reviewer 2: Need to add json5 in our code formatting, will add it in another commit

~~~suggestion
```json5 title=tauri.conf.json5
{
  "build": {
      // devServer URL (comments are allowed!)
      "devPath": "http://localhost:8000",
      "distDir": "../dist"
  }
}
```
~~~

The 1.1.0 release includes TOML support behind the `config-toml` Cargo feature. Now you can define your Tauri configuration in a `Tauri.toml` file:

> ```toml

Reviewer 2: ~~~suggestion title=Tauri.toml
```toml
~~~

[build]
dev-path = "http://localhost:8000"
dist-dir = "../dist"
```

> ### Dependency updates

Reviewer 2: ```suggestion
### Dependency Updates
```

This release includes some dependency updates that must be handled in your app if you implement platform-specific functionalities using these crates. The most important updates are:

- `windows` updated to 0.39.0
- `webview2-com` updated to 0.19.1
- `raw-window-handle` updated to 0.5.0

Make sure you also update plugins such as `window-vibrancy` and `window-shadows` to latest.

## Contributors to 1.1.0

The Tauri team thanks the following contributors for the 1.1.0 release:

- [Berrysoft](https://github.com/Berrysoft)
- [keraf](https://github.com/keraf)
- [jsoref](https://github.com/jsoref)
- [A-kirami](https://github.com/A-kirami)
- [olivierlemasle](https://github.com/olivierlemasle)
- [mateo-gallardo](https://github.com/mateo-gallardo)
- [rockerBOO](https://github.com/rockerBOO)
- [Flecart](https://github.com/Flecart)
- [brian14708](https://github.com/brian14708)
- [koriwi](https://github.com/koriwi)
- [paul-soporan](https://github.com/paul-soporan)
- [shniubobo](https://github.com/shniubobo)
- [horochx](https://github.com/horochx)
- [RubenKelevra](https://github.com/RubenKelevra)
- [LIMPIX31](https://github.com/LIMPIX31)
- [AxlLind](https://github.com/AxlLind)

> # Other changes

Reviewer 2: ```suggestion
# Other Changes
```
